langchain/src/agents/chat_convo/outputParser.ts

```diff
@@ -3,29 +3,67 @@ import { FORMAT_INSTRUCTIONS } from "./prompt.js";
 
 export class ChatConversationalAgentOutputParser extends AgentActionOutputParser {
   async parse(text: string) {
-    let jsonOutput = text.trim();
-    if (jsonOutput.includes("```json")) {
-      jsonOutput = jsonOutput.split("```json")[1].trimStart();
-    } else if (jsonOutput.includes("```")) {
-      const firstIndex = jsonOutput.indexOf("```");
-      jsonOutput = jsonOutput.slice(firstIndex + 3).trimStart();
-    }
-    const lastIndex = jsonOutput.lastIndexOf("```");
-    if (lastIndex !== -1) {
-      jsonOutput = jsonOutput.slice(0, lastIndex).trimEnd();
+    const trimmedText = text.trim();
+    let action: string | undefined;
+    let action_input: string | undefined;
+
+    try {
+      ({ action, action_input } = JSON.parse(trimmedText));
```

Comment:
The agent is prompted to return JSON in a markdown block, so we're actually being optimistic for an invalid response here?